pkg/costmodel/cluster.go

```diff
@@ -715,118 +728,86 @@ func ClusterLoadBalancers(dataSource source.OpenCostDataSource, start, end time.
 	}
 
 	loadBalancerMap := make(map[LoadBalancerIdentifier]*LoadBalancer, len(resActiveMins))
+	activeMap := buildActiveDataMap(resActiveMins, loadBalancerKeyGen, resolution, opencost.NewClosedWindow(start, end))
 
-	for _, result := range resActiveMins {
-		cluster, err := result.GetCluster()
-		if err != nil {
-			cluster = env.GetClusterID()
-		}
-		namespace, err := result.GetNamespace()
-		if err != nil {
-			log.Warnf("ClusterLoadBalancers: LB cost data missing namespace")
-			continue
-		}
-		name, err := result.GetString("service_name")
-		if err != nil {
-			log.Warnf("ClusterLoadBalancers: LB cost data missing service_name")
+	for _, result := range resLBCost {
+		key, ok := loadBalancerKeyGen(result)
+		if !ok {
 			continue
 		}
-		providerID, err := result.GetString("ingress_ip")
-		if err != nil {
-			log.DedupedWarningf(5, "ClusterLoadBalancers: LB cost data missing ingress_ip")
-			providerID = ""
-		}
 
-		key := LoadBalancerIdentifier{
-			Cluster:   cluster,
-			Namespace: namespace,
-			Name:      name,
-		}
+		lbPricePerHr := result.Values[0].Value
 
-		// Skip if there are no data
-		if len(result.Values) == 0 {
-			continue
+		lb := &LoadBalancer{
+			Cluster:   key.Cluster,
+			Namespace: key.Namespace,
+			Name:      key.Name,
+			Cost:      lbPricePerHr, // default to hourly cost, overwrite if active entry exists
+			Ip:        key.IngressIP,
+			Private:   privateIPCheck(key.IngressIP),
 		}
 
-		// Add load balancer to the set of load balancers
-		if _, ok := loadBalancerMap[key]; !ok {
-			loadBalancerMap[key] = &LoadBalancer{
-				Cluster:    cluster,
-				Namespace:  namespace,
-				Name:       fmt.Sprintf("%s/%s", namespace, name), // TODO:ETL this is kept for backwards-compatibility, but not good
-				ProviderID: provider.ParseLBID(providerID),
+		if active, ok := activeMap[key]; ok {
+			lb.Start = active.start
+			lb.End = active.end
+			lb.Minutes = active.minutes
+
+			if active.minutes > 0 {
+				lb.Cost = lbPricePerHr * (active.minutes / 60.0)
```

Comment:
@nikovacevic I think I figured out that there was a presumed error where there were "multiple lbs for a single key" but it happens due to ingress IP changes. I added the ingress ip to the key, and we should no longer get dupes. All of the hand-wavy logic with "scaleFactor" was to account for the multiple duplicates. I replaced with the standard `hourlyPrice * (minutes / 60.0)`

Reply:
My man